tests/fuzz/s2n_deserialize_resumption_state_test.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* Target Functions: s2n_deserialize_resumption_state
+ * This test fuzzes the deserialization logic for our session tickets. This occurs after
+ * the ticket has been decrypted with the server ticket key. Technically it's not
+ * necessary to fuzz decrypted values, however, in the event that an attacker is 
+ * able to get a valid ticket key, we want to make sure these functions can't cause damage.
+ */
+
+#include "api/s2n.h"
+#include "s2n_test.h"
+#include "stuffer/s2n_stuffer.h"
+#include "tls/s2n_resume.h"
+#include "utils/s2n_safety.h"
+
+int s2n_fuzz_test(const uint8_t *buf, size_t len)
+{
+    /* We need at least one byte of input to set parameters */
+    S2N_FUZZ_ENSURE_MIN_LEN(len, 1);
+
+    DEFER_CLEANUP(struct s2n_stuffer fuzzed_ticket = { 0 }, s2n_stuffer_free);
+    POSIX_GUARD(s2n_stuffer_alloc(&fuzzed_ticket, len));
+    POSIX_GUARD(s2n_stuffer_write_bytes(&fuzzed_ticket, buf, len));
+
+    /* Pull a byte off the libfuzzer input and use it to set parameters */
+    uint8_t randval = 0;
+    POSIX_GUARD(s2n_stuffer_read_uint8(&fuzzed_ticket, &randval));
+    POSIX_GUARD(s2n_stuffer_reread(&fuzzed_ticket));
+    POSIX_GUARD(s2n_stuffer_rewrite(&fuzzed_ticket));
+
```

Comment:
Is this reread necessary? Seems like it could lead us to accidentally reusing the same byte of randomness in the future.
And I'd move the rewrite down to where you're setting the first byte, since that's why you're calling it.

Reply:
I think when you write a stuffer the read cursor has to be smaller than the write cursor, no?

Reply:
You're right :( https://github.com/aws/s2n-tls/blob/main/stuffer/s2n_stuffer.c#L39